### ISSUE TITLE
hikey620 recovery fails "remote: invalid partition" when flashing fip.bin

### ISSUE THREAD

**Issue by domdaf** [96boards/documentation#247]
Hello,

I'm trying to recover my Hikey 620 board so I followed the instructions given at  [https://github.com/96boards/documentation/blob/master/ConsumerEdition/HiKey/Installation/BoardRecovery.md]( https://github.com/96boards/documentation/blob/master/ConsumerEdition/HiKey/Installation/BoardRecovery.md) but it always fails when flashing the fastboot  partition.

### Steps:
```
sudo python hisi-idt.py -d /dev/ttyUSB1 --img1=l-loader.bin
+----------------------+
(' Serial: ', '/dev/ttyUSB1')
(' Image1: ', 'l-loader.bin')
(' Image2: ', '')
+----------------------+

('Sending', 'l-loader.bin', '...')
Done

sudo fastboot flash ptable ptable-linux-8g.img
target reported max download size of 268435456 bytes
sending 'ptable' (17 KB)...
OKAY [  0.011s]
writing 'ptable'...
OKAY [  0.015s]
finished. total time: 0.026s
```
Here I wait few seconds
```
sudo fastboot flash fastboot fip.bin
target reported max download size of 268435456 bytes
sending 'fastboot' (1382 KB)...
FAILED (remote: invalid partition)
finished. total time: 0.003s
```
I've tried several times the sequence but there's no way to succeed. Am I doing something wrong? Is my board definitely bricked?  Help is welcome.

Thanks.

**Comment by vchong** [comment]
Did you get `ptable-linux-8g.img` and `fip.bin` from the links in https://github.com/96boards/documentation/blob/master/ConsumerEdition/HiKey/Installation/BoardRecovery.md? Have you tried rebooting and flashing again?

**Comment by domdaf** [comment]
Hi vchong, 
yes I got both files from links in [https://github.com/96boards/documentation/blob/master/ConsumerEdition/HiKey/Installation/BoardRecovery.md](url)

[http://builds.96boards.org/releases/reference-platform/debian/hikey/16.06/bootloader/l-loader.bin](url)

[http://builds.96boards.org/releases/reference-platform/debian/hikey/16.06/bootloader/fip.bin](url)

Yes I have tried several times rebooting and flashing.

I logged the UART output. Don't know if it can help. 

This is what I get after l-loader loading.

NOTICE:  Booting Trusted Firmware
NOTICE:  BL1: v1.1(release):bdec62e
NOTICE:  BL1: Built : 11:11:48, Jun 23 2016
NOTICE:  syspll frequency:1218936832Hz
NOTICE:  succeed to init lpddr3 rank0 dram phy
INFO:    lpddr3_freq_init, set ddrc 533mhz
INFO:    init ddr3 rank0
INFO:    ddr3 rank1 init pass
INFO:    lpddr3_freq_init, set ddrc 800mhz
INFO:    init ddr3 rank0
INFO:    ddr3 rank1 init pass
INFO:    Samsung DDR
NOTICE:  mmc: error status 0x100
NOTICE:  failed to send READY command
NOTICE:  mmc: error status 0x100
NOTICE:  failed to send CMD23
NOTICE:  Enter fastboot mode...
NOTICE:  mmc: error status 0x100
NOTICE:  failed to send command 6
NOTICE:  fail to switch eMMC boot partition
NOTICE:  mmc: error status 0x100
NOTICE:  failed to send CMD23
NOTICE:  Failed to load 'loader_mem' file (-14)
NOTICE:  mmc: error status 0x110
NOTICE:  failed to send CMD25
NOTICE:  Failed to write file (-14)
NOTICE:  mmc: error status 0x100
NOTICE:  failed to send CMD23
NOTICE:  Failed to load 'loader_mem' file (-14)
NOTICE:  mmc: error status 0x100
NOTICE:  failed to send CMD23
NOTICE:  Failed to load 'loader_mem' file (-14)
NOTICE:  mmc: error status 0x100
NOTICE:  failed to send CMD23
NOTICE:  Failed to load 'loader_mem' file (-14)

At this point the fastboot is enable.
```
sudo fastboot devices -l
0123456789ABCDEF       fastboot usb:1-1
```

 I'm able to flash the ptable.
```target reported max download size of 268435456 bytes
sending 'ptable' (17 KB)...
OKAY [  0.011s]
writing 'ptable'...
OKAY [  0.016s]
finished. total time: 0.027s
```

On the UART output I have
NOTICE:  mmc: error status 0x110
NOTICE:  failed to send CMD25
NOTICE:  Failed to write file (-14)
NOTICE:  mmc: error status 0x100
NOTICE:  failed to send CMD23

Then when I try to flash the fastboot I always get
```target reported max download size of 268435456 bytes
sending 'fastboot' (1382 KB)...
FAILED (remote: invalid partition)
finished. total time: 0.003s
```
And no data on the UART output.

**Comment by vchong** [comment]
The UART output helps in the sense that it's showing lots of mmc errors so telling us that the flashing is not working properly, but not sure what's causing them though. The file download links are a bit old, but should still work. Can you maybe try the ones from http://builds.96boards.org/snapshots/reference-platform/components/uefi/latest/hikey/debug/ or http://builds.96boards.org/snapshots/reference-platform/components/uefi-staging/latest/hikey/debug/ just to see if they make any difference? Also, do you have anything else connected to the board, or any changes made to it (e.g. soldering some extra parts or components, etc)?

**Comment by domdaf** [comment]
With [http://builds.96boards.org/snapshots/reference-platform/components/uefi/latest/hikey/debug/](url) I have the same result.

With [http://builds.96boards.org/snapshots/reference-platform/components/uefi-staging/latest/hikey/debug/](url) I can't enter in fastboot mode. In the terminal I just have
```
NOTICE:  Booting Trusted Firmware
NOTICE:  BL1: v1.4(release):v1.4-490-g08e06be8
NOTICE:  BL1: Built : 10:45:57, Jan 11 2018
```
The only change I done on the board was to solder pin header on the J801.

**Comment by vchong** [comment]
Why J801 (UART0)? Aren't you using the default UART3 (J1802) on the LS expansion port for your console? Not sure how accurate it is, but there's a note on http://wiki.lemaker.org/HiKey(LeMaker_version):UART that says connecting VCC to J801-01 will(?) damage the board.

**Comment by domdaf** [comment]
My goal was to  boot a seL4 microkernel on the Hikey 620. For that I followed the steps described in [https://wiki.sel4.systems/Hardware/HiKey](url) that's why I used the UART0. I didn't connect Vcc because I've seen the note on Lemaker wiki. 
I was blocked on wkiki seL4 step #11 (flashing firmware) on the fastboot flashing with invalid partition issue. I thought that something was wrong in the generation of the ptable image so I tried to recover the board with files given in BoardRecovery.md

I've ordered a new board. Hope to receive it next week.
